### PR TITLE
For #25397 - Telemetry for credit card save and update

### PR DESCRIPTION
**Pull request by Alexandru2909**
Record telemetry for credit card autofill prompt save and update interactions.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by Alexandru2909** [comment]
<h1 dir="auto">Request for data collection review form</h1>
<p dir="auto">All questions are mandatory. You must receive review from a
 data steward peer on your responses to these questions before shipping 
new data collection.</p>
<ol dir="auto">
<li>What questions will you answer with this data?</li>
</ol>
<ul dir="auto">
<li>
    Are our users discovering the feature to save from autofill and are they finding it valuable?
</li>
<li>
    How many of our users are saving credit cards from autofill?
</li>
<li>
    How many of our users are updating saved credit cards through autofill?
 </li>
</ul>
<ol dir="auto" start="2">
<li>Why does Mozilla need to answer these questions? Are there benefits 
for users? Do we need this information to address product or business 
requirements?</li>
</ol>
<ul dir="auto">
<li>We need to know how useful credit card autofill is for the users and how 
we can best serve them when it comes to their shopping experiences. This
 data will help us establish a baseline to make product decisions as we 
build out these services. </li>
</ul>
<ol dir="auto" start="3">
<li>What alternative methods did you consider to answer these questions? Why were they not sufficient?</li>
</ol>
<ul dir="auto">
<li>No other alternatives. </li>
</ul>
<ol dir="auto" start="4">
<li>Can current instrumentation answer these questions?</li>
</ol>
<ul dir="auto">
<li>No. </li>
</ul>
<ol dir="auto" start="5">
<li>List all proposed measurements and indicate the category of data 
collection for each measurement, using the Firefox data collection 
categories found on the Mozilla wiki. <br>
Note that the data steward reviewing your request will characterize your
 data collection based on the highest (and most sensitive) category.</li>
</ol>

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Save prompt create: User saved a credit card using the autofill save prompt. | Category 2 - interaction data | [#10343](https://github.com/mozilla-mobile/android-components/issues/10343)
Save prompt update: User update a credit card using the autofill save prompt. | Category 2 - interaction data | [#10343](https://github.com/mozilla-mobile/android-components/issues/10343)

<ol dir="auto" start="6">
<li>Please provide a link to the documentation for this data collection 
which describes the ultimate data set in a public, complete, and 
accurate way.</li>
</ol>
<ul dir="auto">
<li><a href="https://dictionary.telemetry.mozilla.org/apps/fenix" rel="nofollow">https://dictionary.telemetry.mozilla.org/apps/fenix </a></li>
</ul>
<ol dir="auto" start="7">
<li>How long will this data be collected? Choose one of the following:</li>
</ol>
<ul dir="auto">
<li>Up until version 105, with the option to renew at that point.</li>
</ul>
<ol dir="auto" start="8">
<li>What populations will you measure?</li>
</ol>
<ul dir="auto">
<li>All channels, all locales, all countries. </li>
</ul>
<ol dir="auto" start="9">
<li>If this data collection is default on, what is the opt-out mechanism for users?</li>
</ol>
<ul dir="auto">
<li>Default Glean SDK opt-out mechanism. </li>
</ul>
<ol dir="auto" start="10">
<li>Please provide a general description of how you will analyze this data.</li>
</ol>
<ul dir="auto">
<li>Glean. </li>
</ul>
<ol dir="auto" start="11">
<li>Where do you intend to share the results of your analysis?</li>
</ol>
<ul dir="auto">
<li> Only on Glean and with mobile teams. </li>
</ul>
<ol dir="auto" start="12">
<li>Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?</li>
</ol>
<ul dir="auto">
<li>No third-party tools.</li>
</ul>

**Comment by rocketsroger** [comment]
> # Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
>     * Are our users discovering the feature to save from autofill and are they finding it valuable?
> 
>     * How many of our users are saving credit cards from autofill?
> 
>     * How many of our users are updating saved credit cards through autofill?
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits 
>        for users? Do we need this information to address product or business 
>        requirements?
> 
> 
>     * We need to know how useful credit card autofill is for the users and how 
>       we can best serve them when it comes to their shopping experiences. This
>        data will help us establish a baseline to make product decisions as we 
>       build out these services.
> 
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
>     * No other alternatives.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No.
> 
> 
>     5. List all proposed measurements and indicate the category of data 
>        collection for each measurement, using the Firefox data collection 
>        categories found on the Mozilla wiki.    
>        Note that the data steward reviewing your request will characterize your
>         data collection based on the highest (and most sensitive) category.
> 
> 
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Save prompt create: User saved a credit card using the autofill save prompt. 	Category 2 - interaction data 	[#10343](https://github.com/mozilla-mobile/android-components/issues/10343)
> Save prompt update: User update a credit card using the autofill save prompt. 	Category 2 - interaction data 	[#10343](https://github.com/mozilla-mobile/android-components/issues/10343)
> 
>     6. Please provide a link to the documentation for this data collection 
>        which describes the ultimate data set in a public, complete, and 
>        accurate way.
> 
> 
>     * [https://dictionary.telemetry.mozilla.org/apps/fenix   ](https://dictionary.telemetry.mozilla.org/apps/fenix)
> 
> 
>     7. How long will this data be collected? Choose one of the following:
> 
> 
>     * Up until version 105, with the option to renew at that point.
> 
> 
>     8. What populations will you measure?
> 
> 
>     * All channels, all locales, all countries.
> 
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
>     * Default Glean SDK opt-out mechanism.
> 
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean.
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Only on Glean and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
> 
> 
>     * No third-party tools.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by version 105

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

File: app/metrics.yaml [hunk 1]
```diff
@@ -7229,6 +7229,38 @@ credit_cards:
...
+    expires: 105
+    metadata:
+      tags:
+        - Autofill
+  save_prompt_update:
+    type: event
+    description: |
+      User updated a credit card using the autofill save prompt.
+    bugs:
+      - https://github.com/mozilla-mobile/android-components/issues/10343
```
Comment (gabrielluong) [inline app/metrics.yaml:7253]:
```suggestion
      - https://github.com/mozilla-mobile/fenix/issues/25397
```

**Review by rocketsroger — APPROVED** [review]

**Comment by Mugurell** [comment]
`testDebug` failed because of two flaky tests failing:

> SearchFragmentStoreTest. Updating SearchFragmentState from SearchState
> SearchFragmentStoreTest. Updating SearchFragmentState from SearchState - shortcuts disabled

https://github.com/mozilla-mobile/fenix/issues/25170 already exists for that.
I'll rerun just this task.

**Pull request merged**